app/models/article.rb

```diff
@@ -73,4 +77,10 @@ def generate_published_dates
   def downcase_content_format
     self.content_format = self.content_format.downcase
   end
+
+  def create_redirect
+    unless Redirect.exists?(source_path: short_path, target_path: path)
+      Redirect.create(source_path: short_path, target_path: path)
+    end
+  end
```

Comment:
Imagine this situation.

There's a `Redirect` with `source_path: /police`.

Then someone creates an `Article` with the `short_path: police`.

***

I might be reading this wrong, but it looks like it will not create the `Redirect`, but it will create the  `Article` and not raise an error. True?

Seems like we should prevent the `Article` from saving and enforce a unique `short_path` / `Redirect.source_path`. Can you look into that, please?

(Sorry for the delayed review.)